packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -193,7 +208,13 @@ export class SerializedStateManager {
 	 * Updates class snapshot and processedOps if we have a new snapshot and it's among processedOps range.
 	 */
 	private updateSnapshotAndProcessedOpsMaybe() {
-		if (this.latestSnapshot === undefined || this.processedOps.length === 0) {
+		if (
+			this.latestSnapshot === undefined ||
+			this.processedOps.length === 0 ||
+			this.processedOps[this.processedOps.length - 1].sequenceNumber <
+				this.lastSavedOpSequenceNumber ||
+			this.containerDirty()
```

Comment:
I'm not sure we have confidence betting data correctness on "dirty" check.
- Take a look data for fluid:telemetry:ConnectionManager:DesiredConnectionModeMismatch events in Loop telemetry
- Who knows how many bugs like https://github.com/microsoft/FluidFramework/pull/20080 we have and what is their impact (I found it completely by accident)